firebase-firestore/firebase-firestore.gradle

```diff
@@ -144,7 +144,7 @@ dependencies {
     implementation "io.grpc:grpc-protobuf-lite:$grpcVersion"
     implementation "io.grpc:grpc-okhttp:$grpcVersion"
     implementation "io.grpc:grpc-android:$grpcVersion"
-    implementation 'com.google.android.gms:play-services-basement:18.0.0'
+    api 'com.google.android.gms:play-services-basement:18.0.0'
```

Comment:
Sorry, I didn't understand the reason for this change and how it relates to the Flows. Are you  hoping to expose `FirebaseException`?

Reply:
The `firebase-firestore` module exposes `FirebaseException` publicly from `EventListener`:

```kotlin
public interface EventListener<T> {

  /**
   * {@code onEvent} will be called with the new value or the error if an error occurred. It's
   * guaranteed that exactly one of value or error will be non-{@code null}.
   *
   * @param value The value of the event. {@code null} if there was an error.
   * @param error The error if there was error. {@code null} otherwise.
   */
  void onEvent(@Nullable T value, @Nullable FirebaseFirestoreException error);
}
```
Without this, I'm getting this error:

<img width="1180" alt="Screenshot 2022-08-09 at 20 18 52" src="https://user-images.githubusercontent.com/3974977/183731786-22852090-59f3-4366-b3d2-746844bafefb.png">

We could also add `implementation 'com.google.android.gms:play-services-basement:18.0.0'` to the `:firebase-firestore:kts` module but I believe the proposed change above is better as it will fix it for all consumers, even the non-ktx ones.

Reply:
@martinbonnin Ah, I see. Thanks for explaining. 
I would prefer adding that implementation to firebase-firestore-ktx instead of exposing it on the base module.

Reply:
done in https://github.com/firebase/firebase-android-sdk/pull/1252/commits/f6e1aebb9f2d980fd77759fcdcd6e3d4a2bcfb6b